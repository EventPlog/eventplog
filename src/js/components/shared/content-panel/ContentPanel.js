import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  background: white;
  padding: 1rem;
  box-shadow: 1px 2px 4px ${props => lighten(0.45, props.theme.activeLink)};
    
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    border-bottom: 1px solid ${props => lighten(0.2, props.theme.blue)};
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 0.01rem;
    margin: 0 1rem;
    padding-bottom: 5px;
    
    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    margin: 2rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h5 className="content-header">{ title }</h5>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel