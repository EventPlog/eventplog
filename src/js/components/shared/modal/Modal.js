import React from 'react'
import { Button, Header, Icon, Image, Modal } from 'semantic-ui-react'
import styled, {css} from 'styled-components'

const ModalStyles = css`
`
const ModalComponent = ({ className = '', style, children, ...otherProps }) => (
  <Modal className={className}
         style={{ margin: 'auto', marginTop: 'auto', ...style }}
         {...otherProps}>
    {children}
  </Modal>
)

export default styled(ModalComponent)`${ModalStyles}`