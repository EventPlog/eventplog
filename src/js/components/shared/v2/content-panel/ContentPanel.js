import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    margin: 0;
  }

    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    // margin: 2rem 1rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h4 className="content-header">{ title }</h4>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel
