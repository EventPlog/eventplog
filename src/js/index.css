body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

h1, h2, h3, h4, h5, h6 {
  font-family: AndaleMono, "Andale Mono WT", "Lucida Console", "Lucida Sans Typewriter", "DejaVu Sans Mono", "Open Sans";
  color: #a2a3a5;
}

h1, h2, h3 {
  letter-spacing: -1px;
}

h1 {
  font-size: 3rem;
  font-weight: 300;
}

h2 {
  font-size: 2.5rem;
  font-weight: 300;
}

h3 {
  font-size: 2rem;
  font-weight: 500;
}

h4 {
  font-size: 1.8rem;
  font-weight: 500;
}

h5 {
  font-size: 1.5rem;
  font-weight: 500;
}

h6 {
  font-size: 1.2rem;
  font-weight: 500;
}

.app-container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 50px;
}
