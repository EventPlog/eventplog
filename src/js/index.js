import React from 'react';
import ReactDOM from 'react-dom';
import 'semantic-ui-css/semantic.min.css';
import './index.css'
import App from './App.js';
import { BrowserRouter } from 'react-router-dom'
import registerServiceWorker from '../registerServiceWorker';
import { Provider } from 'react-redux'
import configureStore from './store/configureStore'

import { injectGlobal } from 'styled-components';
import { media } from './styles/mixins'
import colors from './styles/theme/variables'

injectGlobal`
  body {
    margin: 0;
    overflow-x: hidden;
	  font-family: 'proxima-nova', sans-serif;
    
  }
  
  .app-container {
    max-width: 1200px;
    margin: 0 auto;
    padding: 0 50px;
    
    ${
      media.phone`
        flex-direction: column;
        padding: 2rem;
      `
    }
  }
  
  *:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
  
  ::selection {
    background: ${ colors.yellow };
  }
  
  ::-moz-selection {
    background: ${ colors.yellow };
  }
  
  p {
    line-height: 1.9rem;
  }
`;


const store = configureStore();

ReactDOM.render(
  <Provider store={store}>
    <BrowserRouter>
      <App store={store} />
    </BrowserRouter>
  </Provider>,
  document.getElementById('root'));

registerServiceWorker();