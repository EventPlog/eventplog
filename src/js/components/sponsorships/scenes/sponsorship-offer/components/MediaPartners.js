import React from 'react'
import { Table, Header, Icon } from 'semantic-ui-react'
import { Follow } from 'react-twitter-widgets'


//======= INTERNAL =========
import ExistingPartnerSearch from 'js/components/partners/scenes/existing-partner-search'
import ContentPanel from 'js/components/shared/content-panel'
const MediaPartners = ({
  event,
  partners = {},
  sponsorships = {},
  deleteSponsorship,
  className = '',
}) => {

  const { data: spData = [] } =  sponsorships
  const sponsorshipsData = spData.filter(s => s.sponsorship_type == 'media')

  const { data = [] } = partners
  const partnerOptions =
    data.map(partner => ({
      key: partner.id,
      value: partner.id,
      text: partner.name,
    }))

  return (
    <ContentPanel className={className} title="Media Partners">
      <Table basic='very' celled collapsing>
        <Table.Body>
          {sponsorshipsData && sponsorshipsData.map(({id, partner = {}}) => (
              <Table.Row>
                <Table.Cell>
                  <Header as='h4' image>
                    <Header.Content>
                      <a href={partner.site_link || `#`}
                         target="_blank">
                        {partner.name}
                      </a>
                      <Header.Subheader>{partner.tagline}</Header.Subheader>
                    </Header.Content>
                  </Header>
                </Table.Cell>
                <Table.Cell>
                  <Follow username={partner.tw_username}/>
                </Table.Cell>
                <Table.Cell>
                  <div class="fb-like"
                       data-href={`https://www.facebook.com/${partner.fb_username}`}
                       data-layout="standard"
                       data-action="like"
                       data-show-faces="true">
                  </div>
                </Table.Cell>

                {event.is_stakeholder &&
                  <Table.Cell>
                    <Icon name="cancel"
                          color="red"
                          size="large"
                          className='btn-delete'
                          onClick={() => deleteSponsorship(id)}
                    />
                  </Table.Cell>
                }
              </Table.Row>
            )
          )}
        </Table.Body>
      </Table>

      {event.is_stakeholder &&
      <ContentPanel className="add-partner" title="Add a media partner">
        <ExistingPartnerSearch sponsorship_type="media" />

      </ContentPanel>
      }

    </ContentPanel>

  )
}

export default MediaPartners