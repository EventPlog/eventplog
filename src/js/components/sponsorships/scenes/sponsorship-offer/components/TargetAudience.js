import React, { Component } from 'react'
import { Table, Header, Checkbox } from 'semantic-ui-react'
import ReactMarkdown from 'react-markdown'

//=========== INTERNAL COMPONENTS ==============
import ContentPanel from 'js/components/shared/content-panel'
import ContentEditable from 'js/components/shared/content-editable'

class TargetAudience extends Component {
  state = { editMode: false }

  toggleEditMode = () => this.setState({ editMode: !this.state.editMode })

  render() {

    const {
      sponsorship_offer,
      handleChange,
      handleSubmit,
      updateEvent,
      event,
      className = ''
    } = this.props

    const { editMode } = this.state
    const canEdit = event.is_stakeholder
    const { topics = [] } = event

    return (
      <ContentPanel className={className} title="Target Audience">
        <Table basic='very' celled collapsing>
          <Table.Body>
            <Table.Row>
              <Table.Cell>
                <Header as='h4' image>
                  <Header.Content>
                    Number of expected guests
                  </Header.Content>
                </Header>
              </Table.Cell>
              <Table.Cell>
                <ContentEditable propName="num_of_expected_guests"
                                 type="input"
                                 canEdit={editMode}
                                 defaultValue={sponsorship_offer.num_of_expected_guests}
                                 onChange={handleChange}
                                 onSubmit={handleSubmit}>
                  {sponsorship_offer.num_of_expected_guests || '100' }
                </ContentEditable>
              </Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>
                <Header as='h4' image>
                  <Header.Content>
                    Age range
                  </Header.Content>
                </Header>
              </Table.Cell>
              <Table.Cell>
                <div className="same-line">
                  <ContentEditable propName="min_age"
                                   type="input"
                                   canEdit={editMode}
                                   defaultValue={sponsorship_offer.min_age}
                                   onChange={handleChange}
                                   onSubmit={handleSubmit}>
                    {sponsorship_offer.min_age || '2'}
                  </ContentEditable>
                  &nbsp;-&nbsp;
                  <ContentEditable propName="max_age"
                                   type="input"
                                   canEdit={editMode}
                                   defaultValue={sponsorship_offer.max_age}
                                   onChange={handleChange}
                                   onSubmit={handleSubmit}>
                    {sponsorship_offer.max_age || '100'}
                  </ContentEditable>
                </div>
              </Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>
                <Header as='h4' image>
                  <Header.Content>
                    Interested in
                  </Header.Content>
                </Header>
              </Table.Cell>
              <Table.Cell>
                <ContentEditable propName="topics"
                                 type="input"
                                 canEdit={editMode}
                                 defaultValue={topics.join(', ')}
                                 onSubmit={() => updateEvent({
                                     id: event.id,
                                     topics: sponsorship_offer.topics
                                   })}
                                 onChange={(key, value) => handleChange(
                                     key,
                                     value.split(',')
                                   )}>
                  {topics.length > 0 ? topics.join(', ') : (canEdit ? 'Add topic tags related to this event (separated by commas)' : '')}
                </ContentEditable>
                <span className="small">
                    Sponsoring this event gives you access to view and send messages to guests.
                  </span>
              </Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>
                <Header as='h4' image>
                  <Header.Content>
                    Pictures from previous events
                  </Header.Content>
                </Header>
              </Table.Cell>
              <Table.Cell>
                <ContentEditable propName="pictures_link"
                                 type="input"
                                 canEdit={editMode}
                                 defaultValue={sponsorship_offer.pictures_link}
                                 onChange={handleChange}
                                 onSubmit={handleSubmit}>
                  <ReactMarkdown
                    source={sponsorship_offer.pictures_link || (canEdit ? 'Click to add a link to your most recent similar event pictures': '')}/>
                </ContentEditable>
              </Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>

        {canEdit &&
          <Checkbox toggle checked={editMode}
                    label={editMode ? "Turn off edit mode" : "Click to edit audience"}
                    onClick={this.toggleEditMode}/>
        }
      </ContentPanel>
    )
  }
}

export default TargetAudience