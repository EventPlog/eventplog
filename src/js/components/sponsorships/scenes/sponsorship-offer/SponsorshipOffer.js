import React from 'react'
import { Icon } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import styled, {css} from 'styled-components'
import ReactMarkdown from 'react-markdown'
import { lighten } from 'polished'

import SponsorshipOfferItems from './SponsorshipOfferItems'
import NewSponsor from '../new-sponsorship-offer-item'
import Button from 'js/components/shared/button'
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import { media } from 'js/styles/mixins'
import Modal from 'js/components/shared/modal'
import ContentEditable from 'js/components/shared/content-editable'
import TargetAudience from './components/TargetAudience'
import MediaPartners from './components/MediaPartners'
import Members from 'js/components/shared/members'
import AboutUser from 'js/components/user/scenes/user/components/AboutUser.js'

import {
  genEventLink,
  genCommunityLink,
  removeSpecialChars
} from 'js/utils'


const StyledSponsorshipOffer = styled.div`
  position: relative;
  background: #634d46;
  
  &.app-container {
    margin-top: 3rem;
    margin-bottom: 6rem;
    padding: 1rem;
    
    > div {
      width: 100%;
    }
    
    ${
      media.phone`
        background: transparent;
        margin-left: -2rem;
        width: calc(100% + 4rem);
        margin-top: 0;
        margin: 0;
      `
    }
  
  }
  
  .pic-frame {
    margin: 2rem 0;
  }
  
  .caption {
    margin: 1rem 0;
  }
  
  img {
    max-width: 100%;
    margin: 1rem auto;
    align-self: center;
  }
  
  .pagination-wrapper {
    display: flex;
  }
  
  .meta-details {
    display: flex;
    flex-direction: column;
    margin: 2rem 2rem 4rem;
    line-height: 1.5;
    
    ${
      media.phone`
        margin-left: 0rem;
        margin-right: 0rem;
      `
    }
    
    form {
      margin-bottom: 2rem;
    }
    
    .field {
      margin-bottom: 2rem;
      
      label {
        font-size: 1.2rem;
      }
    }
  }
  
  button {
    background: var(--yellow);
    border: none;
    color: ${props => props.theme.darkGray};
    font-size: 1.2rem;
  }
  
  .continue-btn {
    background-color: ${props => props.theme.green};
    font-size: 1.2rem;
    padding: 1.2rem;
  }
  
  .pseudo-link {
    color: var(--activeLink);
  }
  
  .content-panel {
    padding-top: 6rem;
    padding-bottom: 4rem;
    background-color: inherit;
    box-shadow: none;
    
    ${
      media.phone`
        padding-right: 2rem;
        padding-left: 2rem;
      `
    }
    
    &.yellow {
      background: #f7f78a;
    }
    
    &.blue {
      background: #b9cbfb;
    }
    
    &.brown {
      background: #fbbba4;
    }
    
    &.purple {
      background: #ceb7fd;
    }
    
    &.dark-purple {
      background: ${props => lighten(-0.05, '#ceb7fd')};
    }
    
    &.gray {
      background: ${props => props.theme.gray};
    }
    
    &.light-green {
      background: #c0dee4;
    }
    
    &.white {
      background: #fff;
    }
    
    .content-header {
      font-size: 2rem;
    }
    
    &.add-partner {
      padding-top: 2rem;
      padding-bottom: 0;
      
      .content-header {
        font-size: 1.2rem;
      }
    }
  }
  
  td .small {
    display: block;
    font-size: 90%;
    color: #888;
  }
  
  .btn-delete {
    cursor: pointer;
    
    &:hover {
      background: ${props => props.theme.gray};
    }
  }
  
  .same-line {
    display: flex;
  }
  
  .sponsorship-package-title {
    align-items: flex-end;
    margin-bottom: 0.5rem;
    display: flex;
    justify-content: space-between;
  }
  
  .edit-cta {
    margin-top: 2rem;
  }
  
  div, p {
    font-size: 1.4rem;
    line-height: 1.6;
  }
  
  .btn-inline {
    display: inline-block;
    margin-top: 1rem;
  }
  
  .avatar-medium {
    margin: 2rem 0;
  }
  
  .custom-sponsor-form {
    padding-top: 2rem;
    padding-bottom: 1rem;
  }
`

export const generateTitle = (event) => {
  return (
    <div className="sponsorship-package-title">
      Sponsorship Packages

      <div className="right">
        { event.is_stakeholder &&
          <Modal style={{padding: '2rem 0'}}
                 trigger={
                    <Button>
                      <Icon name="plus"/>
                      Add
                    </Button>
                  }>
            <ContentPanel title="Add a package you'd like sponsored">
              <NewSponsor newSponsor />
            </ContentPanel>
          </Modal>
        }
      </div>
    </div>
  )
}

export const SponsorshipOffer = ({
  className,
  organizers,
  sponsorships = {},
  sponsorship_offer = {},
  sponsorship_offer_items = {},
  event = {},
  cart,
  loading,
  error,
  currentUser,
  handleChange,
  handleSubmit,
  updateEvent,
  partners = {},
  deleteSponsorship,
  allowNext = () => {}
}) => {

  if (loading) return <Loading />
  if (error) return <Loading.Error msg={error} />

  const { community } = event
  return (
    <StyledSponsorshipOffer className={`${className} sponsors app-container`}>
      <div>
        {(sponsorship_offer.pitch || event.is_stakeholder) &&
          <ContentPanel className="yellow" title="Quick Pitch">
            <ContentEditable propName="pitch"
                             type="textarea"
                             rows="10"
                             style={{minHeight: '10vh'}}
                             canEdit={event.is_stakeholder}
                             defaultValue={sponsorship_offer.pitch}
                             onChange={handleChange}
                             onSubmit={handleSubmit}>
              <ReactMarkdown source={sponsorship_offer.pitch || (event.is_stakeholder ? 'Click to add a quick note convincing people why they should sponsor this event.': '')}/>
            </ContentEditable>
          </ContentPanel>
        }
        {(community || event.is_stakeholder) &&
          <ContentPanel className="gray" title="About the community">
            {community && community.description && <ReactMarkdown source={community.description} />}
            {event.is_stakeholder &&
              <div className="edit-cta">
                <Button.Link className="btn-inline" to={`${genCommunityLink(community)}/edit`}>
                  Update community description
                </Button.Link>
              </div>
            }
          </ContentPanel>
        }

        {(event.goals || event.is_stakeholder) &&
          <ContentPanel className="light-green" title={`About ${event.title}`}>
            <ReactMarkdown source={event.description || event.goals} />

            <img src={event.featured_image} />

            {event.is_stakeholder
              ? <div className="edit-cta">
                  <Button.Link className="btn-inline"
                               to={`${genEventLink(event)}/backstage/settings?activeIndex=1`}>
                    Update event description
                  </Button.Link>
                </div>
              : <div className="edit-cta">
                  <Button.Link target="_blank"
                               inverted
                               className="btn-inline"
                               to={`${genEventLink(event)}`}>
                    View more on event page
                  </Button.Link>
                </div>
            }
          </ContentPanel>
        }

        {organizers &&
          <ContentPanel className="white" title="Meet the organizers">
            <Members>
              {organizers.length > 3
                ? organizers.map(member =>
                    <Members.Member member={member} />
                  )
                : organizers.map(member =>
                    <AboutUser {...{user: member, currentUser }}/>
                  )
              }
            </Members>
            {event.is_stakeholder &&
              <Button.Link className="btn-inline" to={`${genEventLink(event, event.community)}/backstage/settings`}>
                Add more organizers
              </Button.Link>
            }
          </ContentPanel>
        }

        <MediaPartners className="blue" {...{ partners, event, sponsorships, deleteSponsorship }} />

        <TargetAudience className="brown"
                        {...{ sponsorship_offer, handleChange,
                              handleSubmit, updateEvent, event } }/>

        <SponsorshipOfferItems title={generateTitle(event)}
                               currentUser={currentUser}
                               cart={cart}
                               allowNext={allowNext}
                               className="purple"
                               sponsorship_offer_items={sponsorship_offer_items} />

      </div>
    </StyledSponsorshipOffer>
  )
}

export default SponsorshipOffer
