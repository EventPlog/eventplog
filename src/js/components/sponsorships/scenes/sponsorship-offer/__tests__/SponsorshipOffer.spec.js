import React from 'react';
import { SponsorshipOffer, generateTitle } from '../SponsorshipOffer';
import SponsorsList from '../SponsorshipOfferItems';
import TargetAudience from '../components/TargetAudience'
import MediaPartners from '../components/MediaPartners'
import ContentPanel from 'js/components/shared/content-panel'
import NewSponsor from '../../new-sponsorship-offer-item'
import Modal from 'js/components/shared/modal'
import { shallow } from 'enzyme';
import variables from 'js/styles/theme/variables'
import { getUserAvatar } from 'js/utils'

describe('Sponsors', () => {
  const props = {
    event: {id: 3, is_stakeholder: true},
    sponsor: {
      id: 1,
      title: 'An awesome title',
      sponsor_type: 'talk',
      summary: 'A short summary',
      details: 'A lot of details',
      user: {id: 1},
      comments: {
        data: []
      }
    },
    theme: variables,
    currentUser: {id: 1},
    handleViewCount: () => {}
  }

  const wrapper = shallow( <SponsorshipOffer {...props} /> );

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(SponsorsList).length).toEqual(1);
    expect(wrapper.find(MediaPartners).length).toEqual(1);
    expect(wrapper.find(TargetAudience).length).toEqual(1);
  })

  it('renders the right props', () => {
    let SponsorsListInstance = wrapper.find(SponsorsList).at(0)

    expect(SponsorsListInstance.props().title)
        .toEqual(generateTitle(props.event))

    expect(SponsorsListInstance.props().currentUser).toEqual(props.currentUser)
    expect(SponsorsListInstance.props().sponsors).toEqual(props.sponsors)
  })

  describe('when the logged in user is a stakeholder of the event', () => {
    it('shows the pitch content panel', () => {
      expect(wrapper.find(ContentPanel).length).toEqual(2);
      expect(wrapper.find(ContentPanel).at(0).props().title).toEqual('Quick note from organizers');
    })
  })

});