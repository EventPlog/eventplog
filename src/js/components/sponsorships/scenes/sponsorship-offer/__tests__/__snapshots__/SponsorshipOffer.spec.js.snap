// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors should render correctly 1`] = `
<styled.div
  className="undefined sponsors app-container"
>
  <div>
    <ContentPanel
      className="yellow"
      title="Quick Pitch"
    >
      <ContentEditableWrapper
        canEdit={true}
        propName="pitch"
        rows="10"
        style={
          Object {
            "minHeight": "10vh",
          }
        }
        type="textarea"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="Click to add a quick note convincing people why they should sponsor this event."
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </ContentEditableWrapper>
    </ContentPanel>
    <ContentPanel
      className="gray"
      title="About the community"
    >
      <div
        className="edit-cta"
      >
        <Component
          className="btn-inline"
          to="/edit"
        >
          Update community description
        </Component>
      </div>
    </ContentPanel>
    <ContentPanel
      className="light-green"
      title="About undefined"
    >
      <div
        className="edit-cta"
      >
        <Component
          className="btn-inline"
          to="/e/3/backstage/settings?activeIndex=1"
        >
          Update event goals
        </Component>
      </div>
    </ContentPanel>
    <MediaPartners
      className="blue"
      event={
        Object {
          "id": 3,
          "is_stakeholder": true,
        }
      }
      partners={Object {}}
      sponsorships={Object {}}
    />
    <TargetAudience
      className="brown"
      event={
        Object {
          "id": 3,
          "is_stakeholder": true,
        }
      }
      sponsorship_offer={Object {}}
    />
    <SponsorsList
      allowNext={[Function]}
      className="purple"
      currentUser={
        Object {
          "id": 1,
        }
      }
      sponsorship_offer_items={Object {}}
      title={
        <div
          className="sponsorship-package-title"
        >
          Sponsorship Packages
          <div
            className="right"
          >
            <Styled(ModalComponent)
              style={
                Object {
                  "padding": "2rem 0",
                }
              }
              trigger={
                <Button>
                  <Icon
                    as="i"
                    name="plus"
                  />
                  Add
                </Button>
              }
            >
              <ContentPanel
                title="Add a package you'd like sponsored"
              >
                <Unknown
                  newSponsor={true}
                />
              </ContentPanel>
            </Styled(ModalComponent)>
          </div>
        </div>
      }
    />
  </div>
</styled.div>
`;
